tests/middleware/AuthMiddleware.spec.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import { Unauthorized } from '@tsed/exceptions';
+import { assert, expect } from "chai";
+import { AuthMiddleware } from '../../src/middlewares/AuthMiddleware';
+import { AuthService } from '../../src/services/AuthService';
+import * as jwt from 'jsonwebtoken';
+
+describe("AuthMiddleware", () => {
+	const service: AuthMiddleware = new AuthMiddleware(new AuthService());
+	const authService: AuthService = new AuthService();
+
+	let request: any;
+	let response: any;
+	beforeEach(() => {
+		request = {
+			headers: {
+				'Authorization': ''
+			},
+			header: function(header: string):string {
```

Comment:
You could use the same function in this header response as in the one below, because the body of header function is same. Basically, for THIS particular instance, request and response can use same thing:

```
const mockedRequestResponse = { headers: {...}, header: function(...) { ... } }
request = deepcopy(mockedRequestResponse)
response = deepcopy(mockedRequestResponse)
```

The data do not need to be REAL in unit tests. They just have to be real enough.

You've done no mistake in here, it's just for your information that it can be done like that as well. This piece of code is totally fine

Reply:
Oh right! Didn't thought about that. Thanks.